src/Management/test/Endpoint.Test/Actuators/CloudFoundry/CloudFoundrySecurityMiddlewareTest.cs

```diff
@@ -365,6 +384,88 @@ public async Task Redacts_HTTP_headers()
         logMessages.Should().Contain("Authorization: *");
     }
 
+    [InlineData("unavailable", HttpStatusCode.ServiceUnavailable, PermissionsProvider.CloudfoundryNotReachableMessage, MiddlewareUnavailableLog, "true")]
+    [InlineData("unavailable", HttpStatusCode.OK, PermissionsProvider.CloudfoundryNotReachableMessage, MiddlewareUnavailableLog, "false")]
+    [InlineData("not-found", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage, MiddlewareUnauthorizedLog, "true")]
+    [InlineData("not-found", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage, MiddlewareUnauthorizedLog, "false")]
+    [InlineData("unauthorized", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage, MiddlewareUnauthorizedLog, "true")]
+    [InlineData("unauthorized", HttpStatusCode.Unauthorized, PermissionsProvider.InvalidTokenMessage, MiddlewareUnauthorizedLog, "false")]
+    [InlineData("forbidden", HttpStatusCode.Forbidden, PermissionsProvider.AccessDeniedMessage, CFForbiddenLog, "true")]
+    [InlineData("forbidden", HttpStatusCode.Forbidden, PermissionsProvider.AccessDeniedMessage, CFForbiddenLog, "false")]
+    [InlineData("timeout", null, null, CFTimeoutLog, "true")]
+    [InlineData("timeout", null, null, CFTimeoutLog, "false")]
+    [InlineData("exception", HttpStatusCode.InternalServerError, CFExceptionMessage, CFExceptionLog, "true")]
+    [InlineData("exception", HttpStatusCode.OK, CFExceptionMessage, CFExceptionLog, "false")]
+    [InlineData("no_sensitive_data", HttpStatusCode.OK, null, MiddlewareForbiddenLog, "true")]
+    [InlineData("success", HttpStatusCode.OK, null, null, "true")]
+    [Theory]
+    public async Task CloudControllerScenarioTesting(string cloudControllerResponse, HttpStatusCode? steeltoeStatusCode, string? errorMessage,
+        string? expectedLogs, string useStatusCodeFromResponse)
```

Comment:
It took me a while to understand why this works. The HTTP mock abuses the application name to register the different responses. It would be clearer to rename `cloudControllerResponse` to `scenario`.

As _every_ test verifies a scenario, the name `CloudControllerScenarioTesting` should be clarified. Suggestion: Returns expected response on permission check.

Reply:
The change to `InvokeAsyncReturnsExpected` doesn't look right to me. There is no actual `InvokeAsync` call anywhere in the test method body, because it's not a unit test, but an integration test. Therefore, I think we should express the functional scenario being covered in the test name.